rep-0145.rst

```diff
@@ -93,26 +93,50 @@ Topics
 
 The following topics are expected to be common to many devices - an IMU device driver is expected to publish at least one. Note that some of these topics may be also published by support libraries, rather than the base driver implementation. All message types below are supplemented with a std_msgs/Header, containing time and coordinate frame information.
 
+* `imu/data` (sensor_msgs/Imu)
 
-* `imu/data_raw` (sensor_msgs/Imu)
+  - General consolidated sensor data containing `AngularVelocity`, `Attitude`, and `LinearAcceleration` published at the natural frequency of attitude estimation.
 
-  - Sensor output grouping accelerometer (`linear_acceleration`) and gyroscope (`angular_velocity`) data.
+* `imu/angular_velocity` (sensor_msgs/AngularVelocity)
 
-* `imu/data` (sensor_msgs/Imu)
+  - Sensor output containing gyroscope data.
+
+* `imu/attitude` (sensor_msgs/Attitude)
+
+  - Sensor output containing information about the attitude estimation.
 
-  - Same as `imu/data_raw`, with an included quaternion orientation estimate (`orientation`).
+* `imu/linear_acceleration` (sensor_msgs/LinearAcceleration)
+
+  - The sensor output of the measured linear acceleration.
 
 * `imu/mag` (sensor_msgs/MagneticField)
 
   - Sensor output containing magnetometer data.
 
-
-All message types provide a covariance matrix (see REP 103 [1]_) alongside the data field (`*_covariance`). If the data's covariance is unknown, all elements of the covariance matrix should be set to 0, unless overridden by a parameter. If a data field is unreported, the first element (`0`) of the covariance matrix should be set to `-1`.
+All message types provide a covariance matrix (see REP 103 [1]_) alongside the data field (`*_covariance`). If the data's covariance is unknown, all elements of the covariance matrix should be set to 0, unless overridden by a parameter. If a data field is unreported, the first element (`0`) of the covariance matrix should be set to `NaN`.
```

Comment:
I don't quite get this. Should I imagine for example a sensor that can measure only in x-y, but not in z? The last sentence says I should set `_covariance[0]` to `NaN`, but the sensor could actually have a covariance for the x axis, which corresponds to `_covariance[0]`.

Reply:
This was language from the original one - I completely agree on closer examination I don't think this makes sense either.

I think we should change this to be:
- Numbers reported if they got'm (current behavior)
- If **no** information is known, the first element (or all elements) set to NaN (new behavior - in case we have nothing, so we don't even try)
- If a particular piece of information is not known, but others are known, set to `0` (current behavior)

Thoughts?